File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.swift
func solution(_ num:Int) -> Int {
    var count = 0
    var input = num
    
    while input != 1 {
        if count == 500 { return -1 }
        input.isMultiple(of: 2) ? (input /= 2) : (input = (input * 3) + 1)
        count += 1
    }
    
    return count
}

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    long myNum = num;
    int answer = 0;

    if (myNum == 1) {
        return 0;
    }

    while (myNum != 1) {
        if (answer >= 500) {
            return -1;
        }
        if (myNum % 2 == 0) {
            myNum /= 2;
        } else {
            myNum = myNum * 3 + 1;
        }
        answer += 1;
    }

    return answer;
}